src/modules/network/implementation/libp2p-service.js

```diff
@@ -371,22 +372,45 @@ class Libp2pService {
         let readResponseStart;
         let readResponseEnd;
         let response;
+        const timeoutController = new TimeoutController(timeout);
         try {
             readResponseStart = Date.now();
+
+            timeoutController.signal.addEventListener(
+                'abort',
+                async () => {
+                    stream.abort();
+                    response = null;
+                },
+                { once: true },
+            );
+
             response = await this._readMessageFromStream(
                 stream,
                 this.isResponseValid.bind(this),
                 remotePeerId.toB58String(),
             );
+
+            if (timeoutController.signal.aborted) {
+                throw Error('Message timed out!');
```

Comment:
Can we add operation id, peer id, message type to error message?

Reply:
I think that this error is caught at line 402, and there will be logged all informations

Reply:
You are right, thanks